### PR TITLE
ENH: Use ruff for formatting instead of black + update to 0.5.5

### PR DESCRIPTION
**Pull request by theroggy**
- update ruff to 0.5.5
    - apply needed changes in code to comply to new rules
- replace black formatting by ruff-format
    - apply relevant changes in code
    - update contributing docs

Didn't activate `pyupgrade` nor `pydocstyle` yet to keep PR managable.

### CODE REVIEW

File: doc/source/community/contributing.rst [hunk 1]
```diff
@@ -32,11 +32,9 @@ In particular, when submitting a pull request:
   line of a docstring should be a standalone summary.  Parameters and
   return values should be documented explicitly.
 
-- Follow PEP 8 when possible. We use `Black
-  <https://black.readthedocs.io/en/stable/>`_ and `ruff
-  <https://beta.ruff.rs/docs/>`_ to ensure a consistent code
-  format throughout the project. For more details see
-  :ref:`below <contributing_style>`.
+- Follow PEP 8 when possible. We use `ruff <https://beta.ruff.rs/docs/>`_
```
Comment (martinfleis) [inline doc/source/community/contributing.rst:35]:
```suggestion
- Follow PEP 8 when possible. We use `ruff <https://docs.astral.sh/ruff/>`_
```
this is where the old one redirects

File: doc/source/community/contributing.rst [hunk 2]
```diff
@@ -322,24 +320,24 @@ You can find a pull request (or PR) tutorial in the `GitHub's Help Docs <https:/
 Style Guide & Linting
 ---------------------
 
-GeoPandas follows the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_ standard
-and uses `Black <https://black.readthedocs.io/en/stable/>`_ and
-`ruff <https://beta.ruff.rs/docs/>`_ to ensure a consistent code
-format throughout the project.
+GeoPandas follows the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_
+standard and uses `ruff <https://beta.ruff.rs/docs/>`_ to ensure a consistent
```
Comment (martinfleis) [inline doc/source/community/contributing.rst:324]:
```suggestion
standard and uses `ruff <https://docs.astral.sh/ruff/>`_ to ensure a consistent
```

File: doc/source/community/contributing.rst [hunk 3]
```diff
@@ -322,24 +320,24 @@ You can find a pull request (or PR) tutorial in the `GitHub's Help Docs <https:/
...
+code format throughout the project.
 
 Continuous Integration (GitHub Actions) will run those tools and
 report any stylistic errors in your code. Therefore, it is helpful before
 submitting code to run the check yourself::
 
-   black geopandas
+   ruff geopandas
```
Comment (martinfleis) [inline doc/source/community/contributing.rst:331]:
```suggestion
   ruff format geopandas
   ruff check geopandas
```
`ruff <path>` is deprecated and if you're replacing black, you want `ruff format` anyway

File: geopandas/tests/test_extension_array.py [hunk 4]
```diff
@@ -396,7 +396,7 @@ def test_unstack(self, data, index, obj):
 
             expected = obj_ser.unstack(level=level, fill_value=data.dtype.na_value)
             if obj == "series":
-                assert (expected.dtypes == object).all()
+                assert all(is_object_dtype(x) for x in expected.dtypes)
```
Comment (martinfleis) [inline geopandas/tests/test_extension_array.py:399]: Why this change?

Comment (theroggy) [reply geopandas/tests/test_extension_array.py:399]: Ruff gave an E721 linting error on this: https://docs.astral.sh/ruff/rules/type-comparison/

When looking a bit into this, apparently checking the dtype is a bit of a fuss and can give different results depending on how you are doing it, so you need to be careful. Hence, using the function pandas offers to do this seemed like a good idea?

Comment (martinfleis) [reply geopandas/tests/test_extension_array.py:399]: Okay then.

**Review by martinfleis — APPROVED** [review]
Thanks!

CI failures are unrelated afaik.

**Review by m-richards — APPROVED** [review]
Looks fine to me! I don't see a special need to replace black but I suppose it means it's a single consistent tool for linting

**Comment by m-richards** [comment]
Thanks @theroggy!
